lib/puppet/provider/ec2_instance/v2.rb

```diff
@@ -46,10 +46,11 @@ def self.instance_to_hash(region, instance)
     end
     subnet_name = nil
     if instance.subnet_id
-      subnet_response = ec2_client(region).describe_subnets(subnet_ids: [instance.subnet_id])
-      subnet_name_tag = subnet_response.data.subnets.first.tags.detect { |tag| tag.key == 'Name' }
+      subnet_response = ec2_client(region).describe_subnets(filters: [
+        {name: 'subnet-id', values: [instance.subnet_id]}
+      ])
+      subnet_name = subnet_response.data.subnets.empty? ? nil : name_from_tag(subnet_response.data.subnets.first)
```

Comment:
https://tickets.puppetlabs.com/browse/PUP-2643 is an example of what can go wrong when puppet swallows an exception during prefetch. And since PE creates a crontab entry (to run facter periodically) it has generated a PE support escalation. If the aws module fails to retrieve the current state, it should fail hard.